pkg/action/build-iso.go

```diff
@@ -142,19 +136,151 @@ func (b BuildISOAction) prepareISORoot(isoDir string, rootDir string, uefiDir st
 	}
 
 	b.cfg.Logger.Info("Creating EFI image...")
-	err = b.createEFI(uefiDir, filepath.Join(isoDir, constants.IsoEFIPath))
+	err = b.createEFI(rootDir, isoDir)
 	if err != nil {
 		return err
 	}
 	return nil
 }
 
-func (b BuildISOAction) createEFI(root string, img string) error {
-	efiSize, err := utils.DirSize(b.cfg.Fs, root)
+func (b BuildISOAction) createEFI(rootdir string, isoDir string) error {
+	var err error
+	var fallBackShim string
+	// rootfs /efi dir
+	// this is shipped usually with osbuilder and the files come from livecd/grub2-efi-artifacts
+	var fallBackGrub = filepath.Join("/efi", constants.EfiBootPath, "grub.efi")
+	img := filepath.Join(isoDir, constants.IsoEFIPath)
+	temp, _ := utils.TempDir(b.cfg.Fs, "", "enki-iso")
+	utils.MkdirAll(b.cfg.Fs, filepath.Join(temp, constants.EfiBootPath), constants.DirPerm)
+	utils.MkdirAll(b.cfg.Fs, filepath.Join(isoDir, constants.EfiBootPath), constants.DirPerm)
+	// Get possible shim file paths
+	shimFiles := sdk.GetEfiShimFiles(b.cfg.Arch)
+	// Get possible grub file paths
+	grubFiles := sdk.GetEfiGrubFiles(b.cfg.Arch)
+
+	// Calculate shim path based on arch
+	var shimDest string
+	switch b.cfg.Arch {
+	case constants.ArchAmd64, constants.Archx86:
+		shimDest = filepath.Join(temp, constants.ShimEfiDest)
+		fallBackShim = filepath.Join("/efi", constants.EfiBootPath, "bootx64.efi")
+	case constants.ArchArm64:
+		shimDest = filepath.Join(temp, constants.ShimEfiArmDest)
+		fallBackShim = filepath.Join("/efi", constants.EfiBootPath, "bootaa64.efi")
+	default:
+		err = fmt.Errorf("not supported architecture: %v", b.cfg.Arch)
+	}
+
+	shimDone := false
```

Comment:
I don't think it's necessary to be specific and instantiate to false since the zero value of a boolean is always false. But not a big deal if you prefer to ignore since we don't have a code linting specification.

```suggestion
	var shimDone bool
```